File: Tests/Resources/ExpectedTypeScriptCode/Inputs/CustomerInput.ts
import { ID } from "../GraphApi"
import {
  CustomerMarketingOptInLevel,
  TaxExemption,
} from "../Enums"
import { MailingAddressInput } from "./MailingAddressInput"
import { MetafieldInput } from "./MetafieldInput"
import { PrivateMetafieldInput } from "./PrivateMetafieldInput"

export interface CustomerInput {
  acceptsMarketing?: boolean | null;
  acceptsMarketingUpdatedAt?: string | null;
  addresses?: MailingAddressInput[];
  email?: string | null;
  firstName?: string | null;
  id?: ID | null;
  lastName?: string | null;
  locale?: string | null;
  marketingOptInLevel?: CustomerMarketingOptInLevel | null;
  metafields?: MetafieldInput[];
  note?: string | null;
  phone?: string | null;
  privateMetafields?: PrivateMetafieldInput[];
  tags?: string[];
  taxExempt?: boolean | null;
  taxExemptions?: TaxExemption[];
}

File: Tests/Resources/ExpectedKotlinCode/GenerateMultipleFiles/Inputs/CustomerInput.kt

package com.shopify.syrup.inputs

import com.shopify.foundation.syrupsupport.*
import com.shopify.syrup.enums.*
import java.math.BigDecimal
import org.joda.time.DateTime;
import javax.annotation.Generated

@Generated("com.shopify.syrup")
data class CustomerInput(var acceptsMarketing: InputWrapper<Boolean?> = InputWrapper(), var acceptsMarketingUpdatedAt: InputWrapper<DateTime?> = InputWrapper(), var addresses: InputWrapper<List<MailingAddressInput>?> = InputWrapper(), var email: InputWrapper<String?> = InputWrapper(), var firstName: InputWrapper<String?> = InputWrapper(), var id: InputWrapper<ID?> = InputWrapper(), var lastName: InputWrapper<String?> = InputWrapper(), var locale: InputWrapper<String?> = InputWrapper(), var marketingOptInLevel: InputWrapper<CustomerMarketingOptInLevel?> = InputWrapper(), var metafields: InputWrapper<List<MetafieldInput>?> = InputWrapper(), var note: InputWrapper<String?> = InputWrapper(), var phone: InputWrapper<String?> = InputWrapper(), var privateMetafields: InputWrapper<List<PrivateMetafieldInput>?> = InputWrapper(), var tags: InputWrapper<List<String>?> = InputWrapper(), var taxExempt: InputWrapper<Boolean?> = InputWrapper(), var taxExemptions: InputWrapper<List<TaxExemption>?> = InputWrapper())
